packages/ui/src/hooks/useUserTokens.ts

```diff
@@ -0,0 +1,70 @@
+import { paths, setParams } from '@reservoir0x/reservoir-kit-client'
+import { useReservoirClient } from '@reservoir0x/reservoir-kit-ui'
+import useSWRInfinite, { SWRInfiniteConfiguration } from 'swr/infinite'
+
+type UserTokenResponse =
+  paths['/users/{user}/tokens/v5']['get']['responses']['200']['schema']
+
+type UserTokenQuery =
+  paths['/users/{user}/tokens/v5']['get']['parameters']['query']
+
+export default function (
+  user?: string | undefined,
+  options?: UserTokenQuery | false,
+  swrOptions: SWRInfiniteConfiguration = {}
+) {
+  const client = useReservoirClient()
+
+  const { data, mutate, error, isValidating, size, setSize } =
+    useSWRInfinite<UserTokenResponse>(
+      (pageIndex, previousPageData) => {
+        // options are optional, no need to exit early if not provided
+
+        const url = new URL(`${client?.apiBase}/users/${user}/tokens/v4`)
+
+        let query: UserTokenQuery = { ...options }
+
+        if (previousPageData && previousPageData.tokens.length === 0) {
+          // there are no tokens in the previous page so exit b/c previous page was end
+          return null
+        } else if (previousPageData && pageIndex > 0) {
+          query.limit = query.limit + query.offset
```

Comment:
We should set a default limit in case it isn't passed in. Also thinking a bit more about this, the limit is always the same amount (10, 20, etc). It's the offset that changes. So what we need to do here is likely this:
`query.offset = (query?.limit || 20) * pageIndex`
That way, assuming page index 0 is the first page, it'll breakdown to:
```
Page 0 = offset 0
Page 1 = offset 20
Page 2 = offset 40
...
```

Reply:
Good catch!